examples/hybrid/sphere/hadley_circulation.jl

```diff
@@ -0,0 +1,334 @@
+using Test
+using LinearAlgebra
+
+import ClimaCore:
+    ClimaCore,
+    slab,
+    Spaces,
+    Domains,
+    Meshes,
+    Geometry,
+    Topologies,
+    Spaces,
+    Fields,
+    Operators
+import ClimaCore.Utilities: half
+
+using OrdinaryDiffEq: ODEProblem, solve
+using DiffEqBase
+using ClimaTimeSteppers
+
+using NCDatasets, ClimaCoreTempestRemap
+using Statistics: mean
+
+import Logging
+import TerminalLoggers
+Logging.global_logger(TerminalLoggers.TerminalLogger())
+
+# 3D Hadley-like Meridional Circulation (DCMIP 2012 Test 1-2)
+# Reference: http://www-personal.umich.edu/~cjablono/DCMIP-2012_TestCaseDocument_v1.7.pdf, Section 1.2
+
+
+# visualization artifacts
+ENV["GKSwstype"] = "nul"
+using ClimaCorePlots, Plots
+Plots.GRBackend()
+dir = "hadley_circulation"
+path = joinpath(@__DIR__, "output", dir)
+mkpath(path)
+
+function linkfig(figpath, alt = "")
+    # buildkite-agent upload figpath
+    # link figure in logs if we are running on CI
+    if get(ENV, "BUILDKITE", "") == "true"
+        artifact_url = "artifact://$figpath"
+        print("\033]1338;url='$(artifact_url)';alt='$(alt)'\a\n")
+    end
+end
+
+
+# set up function space
+function sphere_3D(
+    R = 6.37122e6,
+    zlim = (0, 12.0e3),
+    helem = 6,
+    zelem = 30,
+    npoly = 4,
+)
+    FT = Float64
+    vertdomain = Domains.IntervalDomain(
+        Geometry.ZPoint{FT}(zlim[1]),
+        Geometry.ZPoint{FT}(zlim[2]);
+        boundary_names = (:bottom, :top),
+    )
+    vertmesh = Meshes.IntervalMesh(vertdomain, nelems = zelem)
+    vert_center_space = Spaces.CenterFiniteDifferenceSpace(vertmesh)
+
+    horzdomain = Domains.SphereDomain(R)
+    horzmesh = Meshes.EquiangularCubedSphere(horzdomain, helem)
+    horztopology = Topologies.Topology2D(horzmesh)
+    quad = Spaces.Quadratures.GLL{npoly + 1}()
+    horzspace = Spaces.SpectralElementSpace2D(horztopology, quad)
+
+    hv_center_space =
+        Spaces.ExtrudedFiniteDifferenceSpace(horzspace, vert_center_space)
+    hv_face_space = Spaces.FaceExtrudedFiniteDifferenceSpace(hv_center_space)
+    return (hv_center_space, hv_face_space)
+end
+
+# RHS tendency specification
+function rhs!(dydt, y, parameters, t, alpha, beta)
```

Comment:
Can we start by caching fields and fixing the allocations?

Reply:
Which allocations do you refer to in particular? And which field to cache? The one for hyperdiffusion?

Reply:
Basically, we shouldn't have anything in the form of

```julia
x = @. y
x = copy(y)
x = similar(y)
```
in the `rhs!` function, as these are allocating functions. Instead, do these outside in global scope, pack them into `parameters`, and unpack them in `rhs!`.

Reply:
Yes. Ok, so if you were referring to the one for hyperdiffusion `ystar`, that stems from the deformation flow example. Can be fixed, but it will need to be fixed also in the other examples

Reply:
Done here and in the deformation flow example as well.